middleware/validator/users.js

```diff
@@ -117,9 +119,48 @@ const resetPassowrd = (req, res, next) => {
   req.password = password;
   return next();
 };
+/**
+ * Function validates user update inputs with customized error messages
+ * @param {Object} req the request body
+ * @param {Object} res the response body
+ * @param {function} next a call to the next function
+ * @returns {undefined}
+ */
+const updateUser = (req, res, next) => {
+  const { decoded } = req.body;
+  // check if all the declared fields are present in 'user' if not,
+  // set the absent variables to undefined
+  const { password, username } = req.body.user ? req.body.user : false;
+  // Validation of user Inputs
+  const validation = validator.updateUserRules({ password });
+  if (validation) {
+    return res.status(400).json({
+      errors: validation,
+    });
+  }
+  return User.findOne({
+    where: { username },
+  })
+    .then((user) => {
+      if (user) {
+        if (user.username === username
+          && user.username !== decoded.username) {
+          return res.status(400)
+            .json({
+              errors: {
+                message: 'Username provided is already taken',
+              },
+            });
+        }
+      }
+      return next();
+    })
+    .catch(next);
+};
 
 export default {
   signup,
   forgotPassword,
-  resetPassowrd,
+  resetPassword,
```

Comment:
All the functions you exported here are middlewares. I think you should make it consistent by using classes and static methods